tracer/test/Datadog.Trace.Tests/Configuration/ServiceNameTests.cs

```diff
@@ -13,69 +14,109 @@ namespace Datadog.Trace.Tests.Configuration
 {
     public class ServiceNameTests
     {
-        private readonly Tracer _tracerV0;
-        private readonly Tracer _tracerV1;
-
-        public ServiceNameTests()
+        private readonly string[] _unmappedKeys = { "elasticsearch", "postgres", "custom-service" };
+        private readonly Dictionary<string, string> _mappings = new()
         {
-            var mappings = new Dictionary<string, string>
-            {
-                { "sql-server", "custom-db" },
-                { "http-client", "some-service" },
-                { "mongodb", "my-mongo" },
-            };
+            { "sql-server", "custom-db" },
+            { "http-client", "some-service" },
+            { "mongodb", "my-mongo" },
+        };
 
-            _tracerV0 = new LockedTracer(new TracerSettings() { MetadataSchemaVersion = SchemaVersion.V0, ServiceNameMappings = mappings });
-            _tracerV1 = new LockedTracer(new TracerSettings() { MetadataSchemaVersion = SchemaVersion.V1, ServiceNameMappings = mappings });
-        }
+        public static IEnumerable<object[]> GetAllConfigs()
+            => from schemaVersion in new object[] { SchemaVersion.V0, SchemaVersion.V1 }
+               from peerServiceTagsEnabled in new[] { true, false }
+               from removeClientServiceNamesEnabled in new[] { true, false }
+               select new[] { schemaVersion, peerServiceTagsEnabled, removeClientServiceNamesEnabled };
 
         [Theory]
-        [InlineData("sql-server", "custom-db")]
-        [InlineData("http-client", "some-service")]
-        [InlineData("mongodb", "my-mongo")]
-        public void RetrievesMappedServiceNames(string serviceName, string expected)
+        [MemberData(nameof(GetAllConfigs))]
+        public void RetrievesMappedServiceNames(object schemaVersionObject, bool peerServiceTagsEnabled, bool removeClientServiceNamesEnabled)
         {
-            _tracerV0.Settings.GetServiceName(_tracerV0, serviceName).Should().Be(expected);
-            _tracerV1.Settings.GetServiceName(_tracerV1, serviceName).Should().Be(expected);
+            var schemaVersion = (SchemaVersion)schemaVersionObject; // Unbox SchemaVersion, which is only defined internally
+            var tracer = new LockedTracer(
+                new TracerSettings()
+                {
+                    MetadataSchemaVersion = schemaVersion,
+                    PeerServiceTagsEnabled = peerServiceTagsEnabled,
+                    RemoveClientServiceNamesEnabled = removeClientServiceNamesEnabled,
+                    ServiceNameMappings = _mappings
+                });
```

Comment:
_sigh_ this is exactly the sort of thing that will be broken by [my PR](https://github.com/DataDog/dd-trace-dotnet/pull/4219) 😞 Will need to switch to a "config-value" approach subsequently

Reply:
Updated in 22fc5b495cb6bbf627e84246e4b433f9fc84b260